File: ts/1024/946.ts
function validateStackSequences(pushed: number[], popped: number[]): boolean {
  const n = pushed.length, stack = []
  for (let i = 0, j = 0; i < n; ++i) {
    stack.push(pushed[i])
    while (stack.length > 0 && j < n && stack[stack.length - 1] === popped[j]) {
      stack.pop()
      j += 1
    }
  }
  return stack.length === 0
};

File: py3/1024/946.py
from typing import List

class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        stack = []
        i, n = 0, len(pushed)
        for v in pushed:
            stack.append(v)
            while stack and i < n and stack[-1] == popped[i]:
                stack.pop()
                i += 1

        return not stack